http-client/src/main/java/io/avaje/http/client/DHttpApi.java

```diff
@@ -34,16 +32,22 @@ <T> void addProvider(Class<T> type, HttpApiProvider<?> apiProvider) {
   }
 
   @SuppressWarnings("unchecked")
-  <T> T provideFor(Class<T> type, HttpClient httpClient) {
-    final var apiProvider = (HttpApiProvider<T>) providerMap.get(type);
+  <T> T provideFor(Class<T> type, HttpClient httpClient, ClassLoader classLoader) {
+    var apiProvider = (HttpApiProvider<T>) providerMap.get(type);
+
+    if (apiProvider == null) {
+      init(classLoader);
```

Comment:
I didn't really like this mutation here with init() ...

... produced 607 as an alternative approach.